sirepo/package_data/static/js/sirepo-components.js

```diff
@@ -581,6 +576,123 @@ SIREPO.app.directive('exportPythonLink', function(appState, panelState) {
     };
 });
 
+SIREPO.app.directive('jobSettingsSbatchLoginAndStartSimulation', function() {
+    return {
+        restrict: 'A',
+        scope: {
+		simState: '<',
+		startSimulation: '&'
+        },
+        template: `
+                <div data-ng-if="simState.showJobSettings()">
+                  <div class="form-group form-group-sm">
+                    <div data-model-field="'jobRunMode'" data-model-name="simState.model" data-label-size="6" data-field-size="6"></div>
+                  </div>
+                  <div data-sbatch-options="simState"></div>
+                </div>
+                <div data-ng-if="needSbatchAgentLogin">
+		<button ng-disabled="loadingSbatchAgentStatus" class="col-sm-6 pull-right btn btn-default" data-ng-click="sbatchLogin()">{{ loginButtonLabel }}</button>
+                </div>
+                <div class="col-sm-6 pull-right" data-ng-if="! needSbatchAgentLogin">
+                  <button class="btn btn-default" ng-disabled="loadingSbatchAgentStatus" data-ng-click="startSimulation()">{{ startButtonLabel }}</button>
+                </div>
+	`,
+        controller: function($rootScope, $scope, appState, authState, errorService, requestSender, stringsService) {
+	    $scope.loadingSbatchAgentStatus = true;
+	    $scope.needSbatchAgentLogin = true;
+            $scope.startButtonLabel = stringsService.startButtonLabel();
+	    setLoginButtonLabel();
+
+	    function requestsbatchAgentStatus() {
+		if (appState.models[$scope.simState.model].jobRunMode !== 'sbatch') {
```

Comment:
When you open the opal visualization page, this line is failing with ```Cannot read properties of undefined (reading 'jobRunMode')```

Reply:
2a5bc9290 . Nice catch. Wasn't expecting errors in codes that didn't use sbatch...